Raven.Database/Bundles/Replication/Controllers/ReplicationController.cs

```diff
@@ -562,7 +578,14 @@ private void SaveReplicationSource(string src, string lastEtag, int batchSize, s
                                  new RavenJObject(), null);
                 });
             }
-
+            //We don't want to delay large batches of attachments otherwise we will endup with alot of data that needs to get replicated.
+            if (Request.Content.Headers.ContentLength > Database.Configuration.Replication.ReplicationPropagationDelaySizeInBytes)
+            {
```

Comment:
Firstly this is a precaution so we don't get a huge batch. 
aggregating received batches is complex and it might have duplicates.
I don't understand what you refer to when you write "We are already doing size limiting when we are replication to destinations"